cli/dts/lib.deno.ns.d.ts

```diff
@@ -163,6 +163,27 @@ declare namespace Deno {
    * */
   export function test(name: string, fn: () => void | Promise<void>): void;
 
+  /** Register a test which will be run when `deno test` is used on the command
+   * line and the containing module looks like a test module.
+   * `fn` can be async if required.
+   *
+   * ```ts
+   * import {assert, fail, assertEquals} from "https://deno.land/std/testing/asserts.ts";
+   * 
+   * Deno.test("My test description", ():void => {
+   *   assertEquals("hello", "hello");
+   * }, { only: true });
+   * 
+   * Deno.test("My async test description", async ():Promise<void> => {
+   *   const decoder = new TextDecoder("utf-8");
+   *   const data = await Deno.readFile("hello_world.txt");
+   *   assertEquals(decoder.decode(data), "Hello world");
+   * }, { only: true });
+   * ```
+   */
+  export function test(name: string, fn: () => void | Promise<void>, options?: TestOptions): void;
+  type TestOptions = Omit<TestDefinition, "name" | "fn">;
```

Comment:
Inline this.

Reply:
@nayeemrmn What do you mean by implementing it?

Reply:
@ohadkoren You need to consume these options in `cli/rt/40_testing.js`. Have you tested that the overload works as expected? You also need to add a test.

Reply:
added a new commit, I'll be happy to get a code review.
Thanks!

Reply:
@nayeemrmn My testing commit has failed, I'll try to fix it.